### PR TITLE
Implemented getBlockchainParameters Method

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java [hunk 1]
```diff
@@ -197,4 +199,34 @@ public void setMaxChunkSizeToHsm(int maxChunkSizeToHsm) {
...
+
+        this.hsmClientProtocol.validatePresenceOf(response, PARAMETERS_FIELD);
+        JsonNode parameters = response.get(PARAMETERS_FIELD);
+
+        this.hsmClientProtocol.validatePresenceOf(parameters, CHECKPOINT_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, MINIMUM_DIFFICULTY_FIELD);
+        this.hsmClientProtocol.validatePresenceOf(parameters, NETWORK_FIELD);
+
+        String checkpoint = parameters.get(CHECKPOINT_FIELD).asText();
+        int minimumDifficulty = parameters.get(MINIMUM_DIFFICULTY_FIELD).asInt();
```
Comment (marcos-iov) [inline src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java:226]: Will this value fit in an `int`? What size can it have?

File: src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java [hunk 2]
```diff
@@ -197,4 +199,34 @@ public void setMaxChunkSizeToHsm(int maxChunkSizeToHsm) {
...
+        String network = parameters.get(NETWORK_FIELD).asText();
+
+        logger.info("[getBlockchainParameters] Checkpoint: {}, Minimum Difficulty: {}, Network: {}", checkpoint, minimumDifficulty, network);
+        return new PowHSMBlockchainParameters(checkpoint, minimumDifficulty, network);
```
Comment (marcos-iov) [inline src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java:230]: Does the checkpoint come with or without the `0x` prefix? Make sure it won't cause a problem when PowHSMBlockchainParameters constructor tries to convert it to Keccak

File: src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java [hunk 3]
```diff
@@ -432,6 +433,38 @@ public void resetAdvanceBlockchain_UnknownError() throws Exception {
         hsmBookkeepingClient.resetAdvanceBlockchain();
     }
 
+    @Test
+    public void getBlockchainParameters_ok() throws JsonRpcException, HSMClientException {
+        Keccak256 expectedCheckpoint = new Keccak256("0000000000000000000000000000000000000000000000000000000000000001");
+        int expectedMinimumDifficulty = 492567908;
```
Comment (marcos-iov) [inline src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java:439]: We should test with a mainnet value, accumulated difficulty for 4000 blocks. Make sure we can handle it

File: src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java (hunk 1 above)
Comment (marcos-iov) [reply src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java:226]: See https://github.com/rsksmart/powpeg-node/blob/hsm3-refactors-integration/src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java#LL32C41-L32C51

Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java]: No It Wont, Refactored It To a BigInteger

File: src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java (hunk 2 above)
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImpl.java:231]: It Comes Without The `0x` And Converts Properly To Keccak

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**